### PR TITLE
PE-3992: upload using credits

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/44f7i0kvrvsog
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/63pti7ulid20g

### CODE REVIEW

**Review by thiagocarvalhodev — COMMENTED** [review]
self review

File: lib/blocs/upload/models/upload_plan.dart [hunk 1]
```diff
@@ -84,19 +83,19 @@ class UploadPlan {
   }
 }
 
-Future<bool> canWeUseTurbo({
```
Comment (thiagocarvalhodev) [inline lib/blocs/upload/models/upload_plan.dart:87]: Remove this code

File: lib/blocs/upload/upload_cubit.dart [hunk 2]
```diff
@@ -462,4 +522,74 @@ class UploadCubit extends Cubit<UploadState> {
...
+      },
+      onFinishFileUpload: (handle) async {
+        handle.writeFileEntityToDatabase(driveDao: _driveDao);
+      },
+      onFinishBundleUpload: (handle) async {
+        try {
+          handle.writeBundleItemsToDatabase(driveDao: _driveDao);
+        } catch (e) {
+          logger.e('Error writing bundle items to database', e);
+        }
```
Comment (karlprieb) [inline lib/blocs/upload/upload_cubit.dart:577]: we don't need the try catch as we will not wait for the Future to resolve. Can you use `unawaited` to make it clear? Do for `onFinishFileUpload` as well

File: lib/blocs/upload/upload_handles/bundle_upload_handle.dart [hunk 3]
```diff
@@ -119,30 +122,7 @@ class BundleUploadHandle implements UploadHandle {
...
-          .upload(bundleTx, maxConcurrentUploadCount: maxConcurrentUploadCount)
-          .map((upload) {
-        uploadProgress = upload.progress;
-        return uploadProgress;
-      });
-    }
-  }
-
-  void dispose() {
+  void dispose({bool useTurbo = false}) {
```
Comment (karlprieb) [inline lib/blocs/upload/upload_handles/bundle_upload_handle.dart:125]: this name is not good. `clearBundleData` is more accurate

**Review by karlprieb — APPROVED** [review]

**Review by matibat — APPROVED** [review]

**Review by matibat — APPROVED** [review]

**Pull request merged**